### PR TITLE
unskip evo2 tests

### PR DESCRIPTION
**Pull request by broland-hat**
### Description

- This PR addresses issue https://github.com/NVIDIA/bionemo-framework/issues/1013
  - In NeMo https://github.com/NVIDIA/NeMo/pull/14515, the code was updated to reduce memory consumption
  - This PR updates the NeMo version to 7ccb0d4.
  - This PR adjusts the memory thresholds to skip tests in sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py
  - This PR adds some tools for torch memory usage.

### Type of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Refactor
- [ ] Documentation update
- [ ] Other (please describe):

### CI Pipeline Configuration

Configure CI behavior by applying the relevant labels:

- [SKIP_CI](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#skip_ci) - Skip all continuous integration tests
- [INCLUDE_NOTEBOOKS_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_notebooks_tests) - Execute notebook validation tests in pytest
- [INCLUDE_SLOW_TESTS](https://github.com/NVIDIA/bionemo-framework/blob/main/docs/docs/user-guide/contributing/contributing.md#include_slow_tests) - Execute tests labelled as slow in pytest for extensive testing

> [!NOTE]
> By default, the notebooks validation tests are skipped unless explicitly enabled.

#### Authorizing CI Runs

We use [copy-pr-bot](https://docs.gha-runners.nvidia.com/apps/copy-pr-bot/#automation) to manage authorization of CI
runs on NVIDIA's compute resources.

- If a pull request is opened by a trusted user and contains only trusted changes, the pull request's code will
  automatically be copied to a pull-request/ prefixed branch in the source repository (e.g. pull-request/123)
- If a pull request is opened by an untrusted user or contains untrusted changes, an NVIDIA org member must leave an
  `/ok to test` comment on the pull request to trigger CI. This will need to be done for each new commit.

### Usage

```python
# TODO: Add code snippet
```

### Pre-submit Checklist

- [x] I have tested these changes locally
- [x] I have updated the documentation accordingly
- [x] I have added/updated tests as needed
- [ ] All existing tests pass successfully

### Local test runs

the slow test **test_evo2.py::test_golden_values_top_k_logits_and_cosine_similarity_7b** is broken on **main**, will be marked skip

- (1 ) test_evo2.py::test_golden_values_top_k_logits_and_cosine_similarity_7b is broken on this commit with **NVIDIA H100 80GB HBM3**
- [pytests_pr1058_unskip_evo2_tests_sub-packages-bionemo-evo2-tests-bionemo-evo2-test_evo2_20250821T0024_6e2a005d.log](https://github.com/user-attachments/files/21922792/pytests_pr1058_unskip_evo2_tests_sub-packages-bionemo-evo2-tests-bionemo-evo2-test_evo2_20250821T0024_6e2a005d.log)
- (2) The same test as in (1) is broken on the commit *424050d2* in main with *NVIDIA H100 80GB HBM3*
[pytests_pr1058_unskip_evo2_tests_sub-packages-bionemo-evo2-tests-bionemo-evo2-test_evo2__test_golden_values_top_k_logits_and_cosine_similarity_7b_20250821T2114_main_424050d2.log](https://github.com/user-attachments/files/21926382/pytests_pr1058_unskip_evo2_tests_sub-packages-bionemo-evo2-tests-bionemo-evo2-test_evo2__test_golden_values_top_k_logits_and_cosine_similarity_7b_20250821T2114_main_424050d2.log)

the slow test **test_evo.py::test_generate_speed** is marked skip per https://nvidia.slack.com/archives/C074Z808N05/p1755185565520729?thread_ts=1755097791.370249&cid=C074Z808N05

### CODE REVIEW

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py [hunk 1]
```diff
@@ -48,6 +48,11 @@
 logger.setLevel(logging.DEBUG)  # Capture all levels in the logger itself
 
 
+MEM_REQUIREMENT_1B_GB = 18  # add 0.6 GB to max mem reserved, and round up
+MEM_REQUIREMENT_7B_GB = 48
```
Comment (jstjohn) [inline sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: so the 7b models used to fit on my A6000, I am surprised it was so close!

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py [hunk 2]
```diff
@@ -724,7 +729,7 @@ def test_generate_speed(
 ):
     is_fp8_supported, compute_capability, device_info = check_fp8_support(torch.cuda.current_device())
     gb_available = torch.cuda.mem_get_info()[0] / 1024**3
-    if (gb_available < 38 and "1b" in ckpt_name) or (gb_available < 50 and "7b" in ckpt_name):
+    if (gb_available < MEM_REQUIREMENT_1B_GB and "1b" in ckpt_name) or (gb_available < MEM_REQUIREMENT_7B_GB and "7b" in ckpt_name):
```
Comment (jstjohn) [inline sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: Can you also modify the py.test.skip message with the variables that you have now. Eg: f"more than {MEM_REQUIREMENT_1B_GB}GB of memory for 1b models"...

Comment (jstjohn) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: Maybe you could pull out all of those identical skip messages into a single variable at the top of the file, or a small function that takes gb_available and returns that message, or something.

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py (hunk 1 above)
Comment (broland-hat) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: ~46.6 + 0.6GB, then rounded up.  0.6GB is the overhead from torch before any tests run.  So this is a estimate, and  pretty close to the threshold for A6000, let me see if there's one of these on computelab.

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py (hunk 2 above)
Comment (broland-hat) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: Sure.

Comment (broland-hat) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: What column width do we use in this repo?

Comment (jstjohn) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: Good question, I don't know off the top of my head... we run `pre_commit`  to apply formatting and I think that should cover checks. You could always break it up if it feels too long regardless...

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py (hunk 1 above)
Comment (jstjohn) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: No worries, I'll check mine and let you know here

Comment (jstjohn) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]:
```
In [5]: gb_available
Out[5]: 41.177490234375

In [6]: torch.cuda.mem_get_info()[1] / 1024**3
Out[6]: 47.4281005859375
```

that's what I have on my A6000 for gb_available vs total. May make sense to just use total. Not sure how available is defined because I'm not running anything at the moment. Yeah... weird... just launching ipython and asking for memory available ate that many gigs at least temporarily.

Comment (jstjohn) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: IMO the total memory is a better thing to skip on... it reflects the hardware rather than the current state of the hardware when running the test. That way something unexpected that leads to memory getting eaten up also leads to test failures rather than skips?

Comment (jstjohn) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: hmm... I have some weird untracked process eating memory, it's 5g used regardless of ipython being ran at the moment.

Comment (broland-hat) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: 5g... that's a lot.

Comment (broland-hat) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: I'm using multi-instance gpu settings, to restrict my 80GB chip to 40GB, and adjust some settings, so I can easily verify these tests.

Comment (broland-hat) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: For me, it looks like startup memory overhead for pytorch (and other libraries) is about 300mb 
image

File: sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py (hunk 2 above)
Comment (broland-hat) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: See item (1) under Local Test Runs in the Description, the test `test_golden_values_top_k_logits_and_cosine_similarity_7b` is failing, and my question would be: is this failing test  due to a code change in the current PR?

Comment (broland-hat) [reply sub-packages/bionemo-evo2/tests/bionemo/evo2/test_evo2.py]: I'm running on H100, compute_capability == "9.0"... this code path may not be tested on CI.
